recipes/pkgconf/all/conanfile.py

```diff
@@ -26,111 +32,90 @@ class PkgConfConan(ConanFile):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
         if not self.options.get_safe("shared", False):
-            tools.replace_in_file(os.path.join(self._source_subfolder, "meson.build"),
+            replace_in_file(self, os.path.join(self.source_folder, "meson.build"),
                                   "'-DLIBPKGCONF_EXPORT'",
                                   "'-DPKGCONFIG_IS_STATIC'")
-        tools.replace_in_file(os.path.join(self._source_subfolder, "meson.build"),
+        replace_in_file(self, os.path.join(self.source_folder, "meson.build"),
             "project('pkgconf', 'c',",
             "project('pkgconf', 'c',\ndefault_options : ['c_std=gnu99'],")
 
     def build(self):
         self._patch_sources()
-        with tools.vcvars(self) if self._is_clang_cl else tools.no_op():
-            meson = self._configure_meson()
-            meson.build()
+        meson = Meson(self)
+        meson.configure()
+        meson.build()
 
     def package(self):
-        self.copy("COPYING", src=self._source_subfolder, dst="licenses")
-        with tools.vcvars(self) if self._is_clang_cl else tools.no_op():
-            meson = self._configure_meson()
-            meson.install()
-
-        if self._is_msvc:
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.pdb")
-            if self.options.enable_lib and not self.options.shared:
-                os.rename(os.path.join(self.package_folder, "lib", "libpkgconf.a"),
-                          os.path.join(self.package_folder, "lib", "pkgconf.lib"),)
-        
-        if not self.options.enable_lib:
-            tools.rmdir(os.path.join(self.package_folder, "lib"))
-            tools.rmdir(os.path.join(self.package_folder, "include"))
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        meson = Meson(self)
+        meson.install()
 
-        tools.rmdir(os.path.join(self.package_folder, "share", "man"))
-        os.rename(os.path.join(self.package_folder, "share", "aclocal"),
-                  os.path.join(self.package_folder, "bin", "aclocal"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        if self.options.enable_lib:
+            fix_apple_shared_install_name(self)
+            fix_msvc_libname(self)
+        else:
+            rmdir(self, os.path.join(self.package_folder, "lib"))
+            rmdir(self, os.path.join(self.package_folder, "include"))
 
-    def package_id(self):
-        if not self.options.enable_lib:
-            del self.info.settings.compiler
+        rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        rmdir(self, os.path.join(self.package_folder, "share", "man"))
+        rename(self, os.path.join(self.package_folder, "share", "aclocal"),
+                     os.path.join(self.package_folder, "bin", "aclocal"))
```

Comment:
Is this just to avoid the hooks? If yes I would like to fix those and added an exception 🙏

Reply:
I think share folder is allowed now by hooks, so it may be avoided.